### PR TITLE
decomp: finish `rolling-obs` | `hud` | `title-obs` | `maincave-obs` | `lavatube-energy`

### PR DESCRIPTION
**Pull request by xTVaser**

### CODE REVIEW

File: goal_src/levels/lavatube/lavatube-energy.gc [hunk 1]
```diff
@@ -5,3 +5,1946 @@
...
+    (if (nonzero? (-> self sound))
+        (stop! (-> self sound))
+        )
+    (none)
+    )
+  :trans
+  (behavior ()
+    (update! (-> self rotation-speed) 0.0)
+    (energyhub-trans)
+    (let ((t9-2 energyhub-set-lava-height))
```
Comment (water111) [inline goal_src/levels/lavatube/lavatube-energy.gc:1600]: I think this function takes a float argument

File: goal_src/levels/rolling/rolling-obs.gc [hunk 2]
```diff
@@ -5,3 +5,1779 @@
...
+  :trans
+  (the-as (function none :behavior pusher) rider-trans)
+  :code
+  (behavior ()
+    (while #t
+      (let ((f0-0 -1.0))
+        (when (and *target* *camera*)
+          (let ((gp-0 (new 'stack-no-clear 'vector)))
+            (vector-! gp-0 (target-pos 0) (camera-pos))
+            (set! f0-0 (ray-capsule-intersect (-> self cyl) (camera-pos) gp-0))
```
Comment (water111) [inline goal_src/levels/rolling/rolling-obs.gc:124]: I think this checks to see if the camera is inside the pusher, and then retracts the pusher out of the way of the camera. Kind of cool

**Review by water111 — CHANGES_REQUESTED** [review]

File: goal_src/levels/rolling/rolling-obs.gc (hunk 2 above)
Comment (xTVaser) [reply goal_src/levels/rolling/rolling-obs.gc:124]: I wonder if you were able to flip the camera around to face jak -- if they'd never get in your way.

File: goal_src/levels/lavatube/lavatube-energy.gc (hunk 1 above)
Comment (xTVaser) [reply goal_src/levels/lavatube/lavatube-energy.gc:1600]: ya you're right, but its unused -- weird.

**Pull request merged**